pkg/bot/socketslack.go

```diff
@@ -519,6 +538,21 @@ func resolveBlockActionCommand(act slack.BlockAction) string {
 		//   @Botkube kcc --verbs get
 		//   @Botkube kcc --resource-type
 		command = fmt.Sprintf("%s %s", act.ActionID, act.SelectedOption.Value)
+	case "plain_text_input":
+		// The main motivation is to extract original command this input belongs to.
+		// a. If the command is located in section blocks, we extract it from section block text
+		// b. If the original command is located inside an input block, we use blockID since the original
+		// command is propagated via that field.
+		blockType := callback.Message.Msg.Blocks.BlockSet[0].BlockType()
+		originalCmd := ""
+		if blockType == slack.MBTSection {
+			originalCmd = callback.Message.Msg.Blocks.BlockSet[0].(*slack.SectionBlock).Text.Text
+		} else if blockType == slack.MBTInput {
+			// blockID contains previously executed command.
+			originalCmd = callback.Message.Msg.Blocks.BlockSet[0].(*slack.InputBlock).BlockID
+		}
```

Comment:
IMO this:
```go
blockType := callback.Message.Msg.Blocks.BlockSet[0].BlockType() // TODO: check blockSet size
originalCmd := ""
if blockType == slack.MBTSection {
	originalCmd = callback.Message.Msg.Blocks.BlockSet[0].(*slack.SectionBlock).Text.Text
} else if blockType == slack.MBTInput {
	// blockID contains previously executed command.
	originalCmd = callback.Message.Msg.Blocks.BlockSet[0].(*slack.InputBlock).BlockID
}
```

can be simplified to this: (it's even more type safe)
```go
block := callback.Message.Msg.Blocks.BlockSet[0] // TODO: check blockSet size
originalCmd := ""
switch b := block.(type) {
case *slack.SectionBlock:
	originalCmd = b.Text.Text
case *slack.InputBlock:
	// blockID contains previously executed command.
	originalCmd = b.BlockID
}
```

Reply:
This suggestion confuses me since I already used `blockID` to propagate original command, but for only pure input actions. For the normal flow where there is no file upload, I already get blocks and use the first one. It contains original command already.

Reply:
discussed f2f :)